### PR TITLE
wip: model has necessary fields/ junit testing for ticket model

### PR DESCRIPTION
**Pull request by elvisEspinozaN**
fields added for controller
junit attempt

### CODE REVIEW

File: src/main/java/genspark/group/boardingpass/model/Ticket.java [hunk 1]
```diff
@@ -44,27 +44,83 @@ public Ticket( Date date,String origin, String destination, Date departureTime,
...
-        Calendar c= Calendar.getInstance();
-        c.setTime(eta);
-        int etaHour= c.get(Calendar.HOUR_OF_DAY); // get hour of eta
-        c.setTime(departureTime);
-        int departureHour= c.get(Calendar.HOUR_OF_DAY); // get hour of departure
-        int flightHours= etaHour-  departureHour;
-        double price= (flightHours* 515)* (1.30); // avg cost/mi, 460 – 575 mph per hour
+        double price= ticketPrice(eta, departureTime);
 
         if(age<= 12) {
```
Comment (spencerk80) [inline src/main/java/genspark/group/boardingpass/model/Ticket.java:125]: This same age discount is also in the method for calculating the age discount. This is being calculated 2x

File: src/main/java/genspark/group/boardingpass/model/Ticket.java [hunk 2]
```diff
@@ -76,45 +132,49 @@ private double generatePrice(int age, String gender, Date eta, Date departureTim
...
+    private double genderDiscount(String gender, Date eta, Date departureTime) { // returns gender discount
+        double price= ticketPrice(eta, departureTime);
+        if(gender.toLowerCase().startsWith("f")) {
+            price*= .25;
+        }
+        return price;
+    }
 
-    public void setAge(int age) {this.age = age;}
+    private double ageDiscount(int age, Date eta, Date departureTime) { // returns age discount
```
Comment (spencerk80) [inline src/main/java/genspark/group/boardingpass/model/Ticket.java:167]: This method is completely disconnected from the price field. The getPrice method will not do anything of these calculations.

File: src/test/java/gentspark/group/boardingpass/dao/TicketTest.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package gentspark.group.boardingpass.dao;
+
+import genspark.group.boardingpass.model.Ticket;
+import org.junit.jupiter.api.Test;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class TicketTest {
```
Comment (spencerk80) [inline src/test/java/gentspark/group/boardingpass/dao/TicketTest.java:9]: Unit testing needs to check all fields and make sure everything is as expected. Dates need to be checked that they are valid, not just null. Is the price with in expected range? etc.

**Review by spencerk80 — CHANGES_REQUESTED** [review]
See various comments viewable in "Files changed"

File: src/main/java/genspark/group/boardingpass/model/Ticket.java [hunk 4]
```diff
@@ -44,27 +44,83 @@ public Ticket( Date date,String origin, String destination, Date departureTime,
...
+
+    public double getGenderDiscount() { // gender discount :
+        return genderDiscount(gender, eta, departureTime);
+    }
+
+    public double getAgeDiscount() { // age discount :
+        return ageDiscount(age, eta, departureTime);
+    }
+
+    public double getPrice() { // total cost :
```
Comment (spencerk80) [inline src/main/java/genspark/group/boardingpass/model/Ticket.java:104]: Unused. Remove

File: src/main/java/genspark/group/boardingpass/model/Ticket.java [hunk 5]
```diff
@@ -11,18 +11,18 @@ Boarding Pass Number, Date, Origin, Destination, Estimated time of arrival (ETA)
...
-    private Date departureTime;
+    private final Date eta;
+    private final Date departureTime;
     private String name;
     private String email;
     private String phoneNumber;
     private String gender;
     private int age;
-    private double price;
+    private final double price;
```
Comment (spencerk80) [inline src/main/java/genspark/group/boardingpass/model/Ticket.java:25]: Unused. Remove

File: src/test/java/gentspark/group/boardingpass/dao/TicketTest.java (hunk 3 above)
Comment (spencerk80) [reply src/test/java/gentspark/group/boardingpass/dao/TicketTest.java:9]: Marking this resolved for this PR. New PR to be opened with unit testing updates

File: src/main/java/genspark/group/boardingpass/model/Ticket.java (hunk 5 above)
Comment (spencerk80) [reply src/main/java/genspark/group/boardingpass/model/Ticket.java:25]: Elvis showed me what's going on and we decided to just name these more clearly. Everything good

**Review by spencerk80 — APPROVED** [review]
Approving for model updates. JUnit tests to come in a later PR

**Review by JackTVanDyke — APPROVED** [review]

**Pull request merged**